api/src/main/java/graphql/nadel/instrumentation/ChainedNadelInstrumentation.java

```diff
@@ -85,13 +86,21 @@ public InstrumentationState createState(NadelInstrumentationCreateStateParameter
     }
 
     @Override
-    public InstrumentationContext<ExecutionResult> beginExecute(NadelInstrumentationExecuteOperationParameters parameters) {
-        return new ChainedInstrumentationContext<>(instrumentations.stream()
+    public CompletableFuture<InstrumentationContext<ExecutionResult>> beginExecute(NadelInstrumentationExecuteOperationParameters parameters) {
+        List<CompletableFuture<InstrumentationContext<ExecutionResult>>> contextFutures = instrumentations.stream()
                 .map(instrumentation -> {
                     InstrumentationState state = getStateFor(instrumentation, parameters.getInstrumentationState());
                     return instrumentation.beginExecute(parameters.withNewState(state));
                 })
-                .collect(toList()));
+                .collect(toList());
+        return sequence(contextFutures).thenApply(ChainedInstrumentationContext::new);
+    }
+
+    private static <T> CompletableFuture<List<T>> sequence(List<CompletableFuture<T>> futures) {
```

Comment:
for the spike just putting this method here to unwrap the list of futures into a list. Probably we can use something from `graphql.execution.Async`?